src/daft-functions/src/list/unique.rs

```diff
@@ -0,0 +1,160 @@
+use std::{any::Any, sync::Arc};
+
+use arrow2::{compute::cast::cast, datatypes::DataType as ArrowDataType, offset::OffsetsBuffer};
+use common_error::{DaftError, DaftResult};
+use daft_core::{
+    array::{growable::make_growable, ListArray},
+    datatypes::{DataType, Field},
+    prelude::Schema,
+    series::{IntoSeries, Series},
+};
+use daft_dsl::{
+    functions::{ScalarFunction, ScalarUDF},
+    ExprRef,
+};
+use serde::{Deserialize, Serialize};
+
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq, Hash)]
+pub struct ListUnique {
+    ignore_nulls: bool,
+}
+
+#[typetag::serde]
+impl ScalarUDF for ListUnique {
+    fn as_any(&self) -> &dyn Any {
+        self
+    }
+
+    fn name(&self) -> &'static str {
+        "list_unique"
+    }
+
+    fn to_field(&self, inputs: &[ExprRef], schema: &Schema) -> DaftResult<Field> {
+        match inputs {
+            [input] => {
+                let field = input.to_field(schema)?;
+                match field.dtype {
+                    DataType::List(inner_type) => {
+                        Ok(Field::new(field.name, DataType::List(inner_type)))
+                    }
+                    DataType::FixedSizeList(inner_type, _) => {
+                        Ok(Field::new(field.name, DataType::List(inner_type)))
+                    }
+                    _ => Err(DaftError::TypeError(format!(
+                        "Expected list input, got {}",
+                        field.dtype
+                    ))),
+                }
+            }
+            _ => Err(DaftError::SchemaMismatch(format!(
+                "Expected 1 input arg, got {}",
+                inputs.len()
+            ))),
+        }
+    }
+
+    fn evaluate(&self, inputs: &[Series]) -> DaftResult<Series> {
```

Comment:
I would move the implementation of evaluate (the stuff inside the match) out of this file and into maybe `daft-core/src/array/ops` to match the other functions. This file is kind of like the definition (naming, typing, etc) of the function, instead of the actual implementation.